lib/jormungandr/test/integration/Cardano/Pool/MetricsSpec.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE TypeApplications #-}
+module Cardano.Pool.MetricsSpec where
+
+import Prelude
+
+import Cardano.BM.Configuration.Static
+    ( defaultConfigStdout )
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Pool.Metrics
+    ( monitorStakePools )
+import Cardano.Wallet.Jormungandr
+    ( toSPBlock )
+import Cardano.Wallet.Jormungandr.Network
+    ( JormungandrBackend (..)
+    , JormungandrConfig (..)
+    , withJormungandr
+    , withNetworkLayer
+    )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..) )
+import Control.Concurrent
+    ( forkIO, killThread, threadDelay )
+import Control.Exception
+    ( throwIO )
+import Control.Monad
+    ( void )
+import Data.Text
+    ( Text )
+import Fmt
+    ( pretty )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import System.Process
+    ( StdStream (..) )
+import Test.Hspec
+    ( Spec, around, it, shouldBe )
+import Test.Integration.Framework.DSL
+    ( eventually )
+
+import qualified Cardano.Pool.DB as Pool
+import qualified Cardano.Pool.DB.Sqlite as Pool
+
+spec :: Spec
+spec = around startNode $ do
+    it "a monitorStakePools thread with db, if killed and restarted, catches up\
+        \ with one that stays running" $ \(nl, referenceDB) ->
+        withDB "/db.sqlite" $ \db -> do
+            referenceWorker <- fork nl referenceDB
+            worker <- fork nl db
+
+            threadDelay (5 * second)
+            killThread worker
+            threadDelay (5 * second)
```

Comment:
These delays are hard-coded. Using say 100 arbitrary delays in sequence wouldn't be feasible, but maybe optimally we could run tests with 100 different delays in parallel.

Reply:
Depends what you're trying to test. If you're trying to test that a killed worker can restart, one could argue you don't even need delays at all. Instead of using delays, you could also simply query the referenceWorker and only restart the second worker when the reference ones has moved on to a next epoch.

Reply:
> If you're trying to test that a killed worker can restart

Yes, hopefully it _can_ restart without crashing. But I think it's much more interesting to know if the restart results in _correct data_. I.e, to let a `worker` operating on `db` be killed for an arbitrary amount of time, and make sure `db == referenceDB` eventually holds (modulo historical stake-distributions).

This way, if the worker were to resume from the wrong cursor, we would know.

Reply:
I'm not sure how to run `monadicIO` in parallel, but I think hard-coded `5 seconds` will be decent. If it is to little the test will trivially pass, not fail unexpectedly.

To comfortably increase the delay to `10s` without making the integration tests run longer I tried to mark this entire test-group as parallel:
```
        describe "No backend required" $ parallel $ do
            describe "Cardano.Wallet.NetworkSpec" $ parallel NetworkLayer.spec
            describe "Mnemonics CLI tests" $ parallel (MnemonicsCLI.spec @t)
            describe "Miscellaneous CLI tests" $ parallel (MiscellaneousCLI.spec @t)
            describe "Launcher CLI tests" $ parallel (LauncherCLI.spec @t)
            describe "Stake Pool Metrics" $ parallel MetricsSpec.spec
```
but I got no improvement 🤔 , so I'm sticking to 5…

Reply:
May you please consider:

> Instead of using delays, you could also simply query the referenceWorker and only restart the second worker when the reference ones has moved on to a next epoch.

We should avoid delays as much possible unless for timeouts. Waiting 5 seconds, or 10 or whatever doesn't tell you much about the underlying state and whether it has changed. In particular, if someone were to change the genesis parameters with different values (:wink:), your delay might become insufficient or wrong. So instead of relying on a delay, rely on a particular event (e.g. the distribution has changed). 

On a different notes, I don't see what parallel test execution has to do with this PR / problem? Plus, there are a few tests in these scenarios that **should not** be ran in parallel. We should be careful about what we mark as parallel. Please, disregard this in this PR.